### PR TITLE
feat: Support proxy config and env vars for HTTP, TCP and TLS Probers

### LINKED ISSUE #193: 能给 TLS Probe Configuration 加上  http proxy 和 自定义 useragent 吗？  谢谢
大大好
      能给 TLS Probe Configuration 加上  http proxy 和 自定义 useragent 吗？ 就和之前的 http Probe Configuration一样， 不知道方便添加不，辛苦，各位大大了。 我们只能享受福利了。

### PR DESCRIPTION
**Pull request by haoel**
close #193 

Code changes:

- Add the `proxy` configuration to support socks5 proxy for TLS and TCP probes.
- Support the `$ALL_PROXY` environment variable for TLS and TCP probes.
- Support the `$HTTP_PROXY` and `$HTTPS_PROXY` environment variable for HTTP probe

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -602,7 +603,8 @@ tcp:
     host: example.com:22
     timeout: 10s # default is 30 seconds
     interval: 2m # default is 60 seconds
-
+    proxy: socks5://proxy.server:1080 # Optional. Only support sock5.
```
Comment (localvar) [inline README.md:606]:
```suggestion
    proxy: socks5://proxy.server:1080 # Optional. Only support socks5.
```

File: README.md [hunk 2]
```diff
@@ -721,6 +723,8 @@ TLS ping to remote endpoint, can probe for revoked or expired certificates
   tls:
     - name: expired test
       host: expired.badssl.com:443
+      proxy: socks5://proxy.server:1080 # Optional. Only support sock5.
```
Comment (localvar) [inline README.md:726]:
```suggestion
      proxy: socks5://proxy.server:1080 # Optional. Only support sock5.
```
```suggestion
      proxy: socks5://proxy.server:1080 # Optional. Only support socks5.
```

File: README.md [hunk 3]
```diff
@@ -964,15 +968,19 @@ notify:
...
-```YAML
-  dry: true # dry notification, print the Discord JSON in log(STDOUT)
-  timeout: 20s # the timeout send out notification, default: 30s
-  retry: # somehow the network is not good and needs to retry.
-    times: 3 # default: 3
-    interval: 10s # default: 5s
-```
+> **Note**:
+>
+> 1) Setting the environment variables - `$HTTP_PROXY` & `$HTTPS_PROX` to configure the proxy for HTTP Webhook notification.
```
Comment (proditis) [inline README.md:973]:
```suggestion
> 1) Setting the environment variables `$HTTP_PROXY` & `$HTTPS_PROXY` allows for configuring the proxy settings for all HTTP related webhook notifications such as discord, slack, telegram etc.
```

**Review by proditis — COMMENTED** [review]
I couldnt get the socks5 to work with environment variables, but i am not sure i tested right. every format combination i tried with ALL_PROXY had no effect. 
HTTP_PROXY and HTTPS_PROXY work as expected each affects only their corresponding protocol.

**Comment by haoel** [comment]
> I couldnt get the socks5 to work with environment variables, but i am not sure i tested right. every format combination i tried with ALL_PROXY had no effect. HTTP_PROXY and HTTPS_PROXY work as expected each affects only their corresponding protocol.

By using the following command, you can create a SOCKS5 proxy. then you can set `ALL_PROXY=socks5://127.0.0.1:1080`

```sh
ssh -D 1080 -qCN user@host.com
```

Besides, you can set a wrong socks5 server, such as `ALL_PROXY=socks5://127.0.0.1:1024`, and you can see the error message `dial tcp 127.0.0.1:1024: connect: connection refused` for TCP or TLS probe.

**Review by localvar — APPROVED** [review]

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**